system/core/document.php

```diff
@@ -581,13 +623,50 @@ function validateRequiredField($name) {
 			return true;
 		}
 
+		/**
+		 * Ensures that the given field, if not empty, contains
+		 * a valid mail address.
+		 *
+		 * Normally, you should set the `type` HTML attribute to
+		 * `email` on the input field to make validate() apply
+		 * this check automatically instead of calling this
+		 * directly.
+		 */
 		function validateEmailField($name) {
 			$value = $this->dataFor($name);
 			if (!$value)
 				return true;
 
 			if (!filter_var($value, FILTER_VALIDATE_EMAIL)) {
-				$this->errors[$name] = __('invalid-email');
+				$this->errors[$name] = __('email-field-invalid');
+				return false;
+			}
+			return true;
+		}
+
+		/**
+		 * Ensures that the given field, if not empty, contains
+		 * no obviously unsafe HTML.
+		 *
+		 * Normally, you should use the `editor` field to make
+		 * validate() apply this check automatically instead of
+		 * calling this directly.
+		 */
+		function validateHtmlField($name) {
+			$value = $this->dataFor($name);
+			if (!$value)
+				return true;
+
+			$dom = $this->root->document()->createElement('dummy');
```

Comment:
Does this need to be cleaned up afterwards?

Reply:
I don't think so, since it is created but not added into the document tree (so I'm assuming PHP will garbage collect the DomNode itself).